diff --git a/cmake/recipes/external/eigen.cmake b/cmake/recipes/external/eigen.cmake
index afd265e3..6dfd1ab7 100644
--- a/cmake/recipes/external/eigen.cmake
+++ b/cmake/recipes/external/eigen.cmake
@@ -8,7 +8,7 @@ include(FetchContent)
 FetchContent_Declare(
     eigen
     GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
-    GIT_TAG tags/3.4.0
+    GIT_TAG tags/5.0.1
     GIT_SHALLOW TRUE
 )
 FetchContent_GetProperties(eigen)
@@ -19,6 +19,8 @@ endif()
 add_library(Eigen3_Eigen INTERFACE)
 add_library(Eigen3::Eigen ALIAS Eigen3_Eigen)
 
+target_compile_definitions(Eigen3_Eigen INTERFACE EIGEN_DONT_VECTORIZE=1)
+
 include(GNUInstallDirs)
 target_include_directories(Eigen3_Eigen SYSTEM INTERFACE
     $<BUILD_INTERFACE:${eigen_SOURCE_DIR}>
diff --git a/include/igl/Singular_Value_Decomposition_Preamble.hpp b/include/igl/Singular_Value_Decomposition_Preamble.hpp
index d55fa86c..2ca5c91b 100644
--- a/include/igl/Singular_Value_Decomposition_Preamble.hpp
+++ b/include/igl/Singular_Value_Decomposition_Preamble.hpp
@@ -37,6 +37,7 @@
 #endif
 
 #ifdef USE_SSE_IMPLEMENTATION
+#include <xmmintrin.h>
 #define ENABLE_SSE_IMPLEMENTATION(X) X
 #else
 #define ENABLE_SSE_IMPLEMENTATION(X)
@@ -55,7 +56,7 @@
 // Alec: Why is this using sse intrinsics if it's supposed to be the scalar
 // implementation?
 #ifdef __SSE__
-#include <mmintrin.h>
+#include <xmmintrin.h>
 // Changed to inline
 inline float rsqrt(const float f)
 {
@@ -74,5 +75,3 @@ inline float rsqrt(const float f)
 }
 #endif
 #endif
-
-
diff --git a/include/igl/active_set.cpp b/include/igl/active_set.cpp
index 0ba567ee..f127a0f4 100644
--- a/include/igl/active_set.cpp
+++ b/include/igl/active_set.cpp
@@ -280,7 +280,7 @@ IGL_INLINE igl::SolverStatus igl::active_set(
       cout<<"  everything's fixed."<<endl;
 #endif
       Z.resize(A.rows(),Y_i.cols());
-      Z(known_i,Eigen::all) = Y_i;
+      Z(known_i,Eigen::indexing::all) = Y_i;
       sol.resize(0,Y_i.cols());
       assert(Aeq_i.rows() == 0 && "All fixed but linearly constrained");
     }else
@@ -322,7 +322,7 @@ IGL_INLINE igl::SolverStatus igl::active_set(
     // Slow
     slice(A,known_i,1,Ak);
     //slice(B,known_i,Bk);
-    DerivedB Bk = B(known_i,Eigen::all);
+    DerivedB Bk = B(known_i,Eigen::indexing::all);
     MatrixXd Lambda_known_i = -(0.5*Ak*Z + 0.5*Bk);
     // reverse the lambda values for lx
     Lambda_known_i.block(nk,0,as_lx_count,1) =
diff --git a/include/igl/bijective_composite_harmonic_mapping.cpp b/include/igl/bijective_composite_harmonic_mapping.cpp
index 8b253d99..1df20096 100644
--- a/include/igl/bijective_composite_harmonic_mapping.cpp
+++ b/include/igl/bijective_composite_harmonic_mapping.cpp
@@ -50,7 +50,7 @@ IGL_INLINE bool igl::bijective_composite_harmonic_mapping(
   assert(F.cols() == 3 && "F should contain triangles");
   int nsteps = min_steps;
   Eigen::Matrix<typename Derivedbc::Scalar, Eigen::Dynamic, Eigen::Dynamic> bc0 =
-    V(b.col(0),Eigen::all);
+    V(b.col(0),Eigen::indexing::all);
 
   // It's difficult to check for flips "robustly" in the sense that the input
   // mesh might not have positive/consistent sign to begin with.
@@ -81,7 +81,7 @@ IGL_INLINE bool igl::bijective_composite_harmonic_mapping(
         //mw.save(bct,"bct");
         //mw.write("numerical.mat");
         harmonic(Eigen::Matrix<typename DerivedU::Scalar, Eigen::Dynamic, Eigen::Dynamic>(U), F, b, bct, 1, U);
-        bct = U(b.col(0),Eigen::all);
+        bct = U(b.col(0),Eigen::indexing::all);
         nans = (U.array() != U.array()).count();
         if(test_for_flips)
         {
diff --git a/include/igl/blue_noise.cpp b/include/igl/blue_noise.cpp
index c2c49caa..05db8e64 100644
--- a/include/igl/blue_noise.cpp
+++ b/include/igl/blue_noise.cpp
@@ -289,10 +289,10 @@ IGL_INLINE void igl::blue_noise(
   {
     Eigen::VectorXi I;
     igl::sortrows(decltype(Xs)(Xs),true,Xs,I);
-    X = X(I,Eigen::all).eval();
+    X = X(I,Eigen::indexing::all).eval();
     // These two could be spun off in their own thread.
-    XB = XB(I,Eigen::all).eval();
-    XFI = XFI(I,Eigen::all).eval();
+    XB = XB(I,Eigen::indexing::all).eval();
+    XFI = XFI(I,Eigen::indexing::all).eval();
   }
   // Initialization
   std::unordered_map<BlueNoiseKeyType,std::vector<int> > M;
diff --git a/include/igl/copyleft/cgal/minkowski_sum.cpp b/include/igl/copyleft/cgal/minkowski_sum.cpp
index 0741e4b7..a66c50a1 100644
--- a/include/igl/copyleft/cgal/minkowski_sum.cpp
+++ b/include/igl/copyleft/cgal/minkowski_sum.cpp
@@ -230,22 +230,22 @@ IGL_INLINE void igl::copyleft::cgal::minkowski_sum(
   typedef Matrix<typename DerivedG::Scalar,Dynamic,1> VectorXI;
   MatrixXI GT(mp+mn,3);
   GT<< 
-    FA(igl::find(N),Eigen::all), 
-    (FA.array()+n).eval()(igl::find(P),Eigen::all);
+    FA(igl::find(N),Eigen::indexing::all), 
+    (FA.array()+n).eval()(igl::find(P),Eigen::indexing::all);
 
   // J indexes FA for parts at s and m+FA for parts at d
   J.derived() = igl::LinSpaced<DerivedJ >(m,0,m-1);
   DerivedJ JT(mp+mn);
   JT << 
-    J(igl::find(P),Eigen::all), 
-    J(igl::find(N),Eigen::all);
+    J(igl::find(P),Eigen::indexing::all), 
+    J(igl::find(N),Eigen::indexing::all);
   JT.block(mp,0,mn,1).array()+=m;
 
   // Original non-co-planar faces with positively oriented reversed
   MatrixXI BA(mp+mn,3);
   BA << 
-    FA(igl::find(P),Eigen::all).rowwise().reverse(), 
-    FA(igl::find(N),Eigen::all);
+    FA(igl::find(P),Eigen::indexing::all).rowwise().reverse(), 
+    FA(igl::find(N),Eigen::indexing::all);
   // Quads along **all** sides
   MatrixXI GQ((mp+mn)*3,4);
   GQ<< 
diff --git a/include/igl/copyleft/cgal/point_areas.cpp b/include/igl/copyleft/cgal/point_areas.cpp
index 36313f8c..7113bdef 100644
--- a/include/igl/copyleft/cgal/point_areas.cpp
+++ b/include/igl/copyleft/cgal/point_areas.cpp
@@ -68,17 +68,17 @@ namespace igl {
         igl::parallel_for(P.rows(),[&](int i)
         {
           MatrixP neighbors;
-          neighbors = P(I.row(i),Eigen::all);
+          neighbors = P(I.row(i),Eigen::indexing::all);
           if(N.rows() && neighbors.rows() > 1){
             MatrixN neighbor_normals;
-            neighbor_normals = N(I.row(i),Eigen::all);
+            neighbor_normals = N(I.row(i),Eigen::indexing::all);
             Eigen::Matrix<scalarN,1,3> poi_normal = neighbor_normals.row(0);
             Eigen::Matrix<scalarN,Eigen::Dynamic,1> dotprod =
                             poi_normal(0)*neighbor_normals.col(0)
             + poi_normal(1)*neighbor_normals.col(1)
             + poi_normal(2)*neighbor_normals.col(2);
             Eigen::Array<bool,Eigen::Dynamic,1> keep = dotprod.array() > 0;
-            neighbors = neighbors(igl::find(keep),Eigen::all).eval();
+            neighbors = neighbors(igl::find(keep),Eigen::indexing::all).eval();
           }
           if(neighbors.rows() <= 2){
             A(i) = 0;
@@ -96,7 +96,7 @@ namespace igl {
               T.row(i) *= -1;
             }
             
-            MatrixP plane = scores(Eigen::all,{0,1});
+            MatrixP plane = scores(Eigen::indexing::all,{0,1});
             
             std::vector< std::pair<Point,unsigned> > points;
             //This is where we obtain a delaunay triangulation of the points
diff --git a/include/igl/copyleft/cgal/point_solid_signed_squared_distance.cpp b/include/igl/copyleft/cgal/point_solid_signed_squared_distance.cpp
index 3ce005b1..4cbee707 100644
--- a/include/igl/copyleft/cgal/point_solid_signed_squared_distance.cpp
+++ b/include/igl/copyleft/cgal/point_solid_signed_squared_distance.cpp
@@ -33,7 +33,7 @@ IGL_INLINE void igl::copyleft::cgal::point_solid_signed_squared_distance(
   // Collect queries that have non-zero distance
   Eigen::Array<bool,Eigen::Dynamic,1> NZ = D.array()!=0;
   // Compute sign for non-zero distance queries
-  DerivedQ QNZ = Q(igl::find(NZ),Eigen::all);
+  DerivedQ QNZ = Q(igl::find(NZ),Eigen::indexing::all);
   Eigen::Array<bool,Eigen::Dynamic,1> DNZ;
   igl::copyleft::cgal::points_inside_component(VB,FB,QNZ,DNZ);
   // Apply sign to distances
diff --git a/include/igl/copyleft/cgal/trim_with_solid.cpp b/include/igl/copyleft/cgal/trim_with_solid.cpp
index 0e81a35c..d9e80ea8 100644
--- a/include/igl/copyleft/cgal/trim_with_solid.cpp
+++ b/include/igl/copyleft/cgal/trim_with_solid.cpp
@@ -165,7 +165,7 @@ IGL_INLINE void igl::copyleft::cgal::trim_with_solid(
       igl::copyleft::cgal::intersect_other(
         VA,FA,VB,FB,{false,false,true},_1,V,F,J,_2);
       const auto keep = igl::find( (J.array()<FA.rows()).eval() );
-      F = F(keep,Eigen::all).eval();
+      F = F(keep,Eigen::indexing::all).eval();
       J = J(keep).eval();
       {
         Eigen::VectorXi _;
@@ -262,7 +262,7 @@ IGL_INLINE void igl::copyleft::cgal::trim_with_solid(
                 keep.push_back(f);
               }
             }
-            F = F(keep,Eigen::all).eval();
+            F = F(keep,Eigen::indexing::all).eval();
             J = J(keep).eval();
           }
 
@@ -278,7 +278,7 @@ IGL_INLINE void igl::copyleft::cgal::trim_with_solid(
       // only keep faces from A
       Eigen::Array<bool,Eigen::Dynamic,1> A = J.array()< FA.rows();
       const auto AI = igl::find(A);
-      F = F(AI,Eigen::all).eval();
+      F = F(AI,Eigen::indexing::all).eval();
       J = J(AI).eval();
       P = P(AI).eval();
       set_D_via_patches(num_patches,P);
diff --git a/include/igl/copyleft/cgal/wire_mesh.cpp b/include/igl/copyleft/cgal/wire_mesh.cpp
index 393d2c66..1cb5537d 100644
--- a/include/igl/copyleft/cgal/wire_mesh.cpp
+++ b/include/igl/copyleft/cgal/wire_mesh.cpp
@@ -115,7 +115,7 @@ IGL_INLINE void igl::copyleft::cgal::wire_mesh(
   const auto append_hull = 
     [&V,&vF,&vJ](const Eigen::VectorXi & I, const int j)
   {
-    MatrixX3S Vv = V(I,Eigen::all);
+    MatrixX3S Vv = V(I,Eigen::indexing::all);
 
     if(coplanar(Vv))
     {
diff --git a/include/igl/decimate.cpp b/include/igl/decimate.cpp
index 8597cf48..af16e875 100644
--- a/include/igl/decimate.cpp
+++ b/include/igl/decimate.cpp
@@ -68,7 +68,7 @@ IGL_INLINE bool igl::decimate(
     J,
     I);
   const Eigen::Array<bool,Eigen::Dynamic,1> keep = (J.array()<orig_m);
-  G = G(igl::find(keep),Eigen::all).eval();
+  G = G(igl::find(keep),Eigen::indexing::all).eval();
   J = J(igl::find(keep)).eval();
   Eigen::VectorXi _1,I2;
   igl::remove_unreferenced(Eigen::MatrixXd(U),Eigen::MatrixXi(G),U,G,_1,I2);
diff --git a/include/igl/ears.cpp b/include/igl/ears.cpp
index 3173c2bb..6f3bb6b1 100644
--- a/include/igl/ears.cpp
+++ b/include/igl/ears.cpp
@@ -21,7 +21,7 @@ IGL_INLINE void igl::ears(
   }
   find((B.rowwise().count() == 2).eval(), ear);
   // Why do I need this .derived()?
-  Eigen::Array<bool, Eigen::Dynamic, 3> Bear = B(ear.derived(),Eigen::all);
+  Eigen::Array<bool, Eigen::Dynamic, 3> Bear = B(ear.derived(),Eigen::indexing::all);
   Eigen::Array<bool, Eigen::Dynamic, 1> M;
   igl::min(Bear,2,M,ear_opp);
 }
diff --git a/include/igl/eigs.cpp b/include/igl/eigs.cpp
index ac3da97a..6d0e4cd2 100644
--- a/include/igl/eigs.cpp
+++ b/include/igl/eigs.cpp
@@ -159,7 +159,7 @@ IGL_INLINE bool igl::eigs(
   // finally sort
   VectorXi I;
   igl::sort(S,1,false,sS,I);
-  sU = U(Eigen::all,I);
+  sU = U(Eigen::indexing::all,I);
   sS /= rescale;
   sU /= sqrt(rescale);
   return true;
diff --git a/include/igl/iterative_closest_point.cpp b/include/igl/iterative_closest_point.cpp
index c4c9b4b4..57f90caf 100644
--- a/include/igl/iterative_closest_point.cpp
+++ b/include/igl/iterative_closest_point.cpp
@@ -96,7 +96,7 @@ IGL_INLINE void igl::iterative_closest_point(
       Ytree.squared_distance(VY,FY,X,sqrD,I,P);
     }
     // Use better normals?
-    MatrixXS N = NY(I,Eigen::all);
+    MatrixXS N = NY(I,Eigen::indexing::all);
     //MatrixXS N = (X - P).rowwise().normalized();
     // fit rotation,translation
     Matrix3S Rup;
diff --git a/include/igl/linprog.cpp b/include/igl/linprog.cpp
index e30ed531..8181b322 100644
--- a/include/igl/linprog.cpp
+++ b/include/igl/linprog.cpp
@@ -76,7 +76,7 @@ IGL_INLINE bool igl::linprog(
     J.head(j) = B.head(j);
     J.tail(B.size()-j-1) = B.tail(B.size()-j-1);
     B(j) = n+m;
-    MatrixXd AJ = A(Eigen::all,J);
+    MatrixXd AJ = A(Eigen::indexing::all,J);
     const VectorXd a = b - AJ.rowwise().sum();
     {
       MatrixXd old_A = A;
@@ -123,7 +123,7 @@ IGL_INLINE bool igl::linprog(
       }
       // reduced costs
       VectorXd sN = s(N);
-      MatrixXd AN = A(Eigen::all,N);
+      MatrixXd AN = A(Eigen::indexing::all,N);
       VectorXd r = sN - AN.transpose() * yb;
       int q;
       // determine new basic variable
@@ -209,7 +209,7 @@ IGL_INLINE bool igl::linprog(
       }
     }
     // iterative refinement
-    xb = (xb+D*(b-A(Eigen::all,B)*xb)).eval();
+    xb = (xb+D*(b-A(Eigen::indexing::all,B)*xb)).eval();
     // must be due to rounding
     VectorXi I;
     igl::find((xb.array()<0).eval(),I);
diff --git a/include/igl/min_quad_with_fixed.impl.h b/include/igl/min_quad_with_fixed.impl.h
index 48da2570..7d9b4904 100644
--- a/include/igl/min_quad_with_fixed.impl.h
+++ b/include/igl/min_quad_with_fixed.impl.h
@@ -440,7 +440,7 @@ IGL_INLINE bool igl::min_quad_with_fixed_solve(
     }
 
     // Build right hand side
-    MatrixXT BBequlcols = BBeq(data.unknown_lagrange,Eigen::all);
+    MatrixXT BBequlcols = BBeq(data.unknown_lagrange,Eigen::indexing::all);
     MatrixXT NB;
     if(kr == 0)
     {
@@ -489,7 +489,7 @@ IGL_INLINE bool igl::min_quad_with_fixed_solve(
       //data.AeqTQR.colsPermutation().transpose() * (-data.Aeqk * Y + Beq);
       data.AeqTET * (-data.Aeqk * Y + Beq.replicate(1,Beq.cols()==cols?1:cols));
     // Where did this -0.5 come from? Probably the same place as above.
-    MatrixXT Bu = B(data.unknown,Eigen::all);
+    MatrixXT Bu = B(data.unknown,Eigen::indexing::all);
     MatrixXT NB;
     NB = -0.5*(Bu.replicate(1,B.cols()==cols?1:cols) + data.preY * Y);
     // Trim eff_Beq
diff --git a/include/igl/normal_derivative.cpp b/include/igl/normal_derivative.cpp
index d020c790..11e2c114 100644
--- a/include/igl/normal_derivative.cpp
+++ b/include/igl/normal_derivative.cpp
@@ -40,7 +40,7 @@ IGL_INLINE void igl::normal_derivative(
     case 4:
     {
       const MatrixXi DDJ = 
-        Ele(Eigen::all,{1,0,2,0,3,0,2,1,3,1,0,1,3,2,0,2,1,2,0,3,1,3,2,3});
+        Ele(Eigen::indexing::all,{1,0,2,0,3,0,2,1,3,1,0,1,3,2,0,2,1,2,0,3,1,3,2,3});
       MatrixXi DDI(m,24);
       for(size_t f = 0;f<4;f++)
       {
@@ -53,7 +53,7 @@ IGL_INLINE void igl::normal_derivative(
       const DiagonalMatrix<Scalar,24,24> S =
         (Matrix<Scalar,2,1>(1,-1).template replicate<12,1>()).asDiagonal();
       Matrix<Scalar,Dynamic,Dynamic> DDV =
-        C(Eigen::all,{2,2,1,1,3,3,0,0,4,4,2,2,5,5,1,1,0,0,3,3,4,4,5,5});
+        C(Eigen::indexing::all,{2,2,1,1,3,3,0,0,4,4,2,2,5,5,1,1,0,0,3,3,4,4,5,5});
       DDV *= S;
 
       IJV.reserve(DDV.size());
@@ -70,7 +70,7 @@ IGL_INLINE void igl::normal_derivative(
     }
     case 3:
     {
-      const MatrixXi DDJ = Ele(Eigen::all,{2,0,1,0,0,1,2,1,1,2,0,2});
+      const MatrixXi DDJ = Ele(Eigen::indexing::all,{2,0,1,0,0,1,2,1,1,2,0,2});
       MatrixXi DDI(m,12);
       for(size_t f = 0;f<3;f++)
       {
@@ -82,7 +82,7 @@ IGL_INLINE void igl::normal_derivative(
       }
       const DiagonalMatrix<Scalar,12,12> S =
         (Matrix<Scalar,2,1>(1,-1).template replicate<6,1>()).asDiagonal();
-      Matrix<Scalar,Dynamic,Dynamic> DDV = C(Eigen::all,{1,1,2,2,2,2,0,0,0,0,1,1});
+      Matrix<Scalar,Dynamic,Dynamic> DDV = C(Eigen::indexing::all,{1,1,2,2,2,2,0,0,0,0,1,1});
       DDV *= S;
 
       IJV.reserve(DDV.size());
diff --git a/include/igl/predicates/polygons_to_triangles.cpp b/include/igl/predicates/polygons_to_triangles.cpp
index 1db6c2c4..1eea9d40 100644
--- a/include/igl/predicates/polygons_to_triangles.cpp
+++ b/include/igl/predicates/polygons_to_triangles.cpp
@@ -54,7 +54,7 @@ IGL_INLINE void igl::predicates::polygons_to_triangles(
               Eigen::Vector3d _1;
               Eigen::Vector3i I;
               igl::sort(es.eigenvalues().real().eval(),1,false,_1,I);
-              C = C(Eigen::all,I).eval();
+              C = C(Eigen::indexing::all,I).eval();
             }
             S = P*C.leftCols(2);
             break;
diff --git a/include/igl/qslim.cpp b/include/igl/qslim.cpp
index 4235f05f..4dbfb412 100644
--- a/include/igl/qslim.cpp
+++ b/include/igl/qslim.cpp
@@ -74,7 +74,7 @@ IGL_INLINE bool igl::qslim(
   // Remove phony boundary faces and clean up
   const Eigen::Array<bool,Eigen::Dynamic,1> keep = (J.array()<orig_m);
   const auto keep_i = igl::find(keep);
-  G = G(keep_i,Eigen::all).eval();
+  G = G(keep_i,Eigen::indexing::all).eval();
   J = J(keep_i).eval();
   Eigen::VectorXi _1,I2;
   igl::remove_unreferenced(Eigen::MatrixXd(U),Eigen::MatrixXi(G),U,G,_1,I2);
diff --git a/include/igl/ramer_douglas_peucker.cpp b/include/igl/ramer_douglas_peucker.cpp
index cd3577bb..91e575ff 100644
--- a/include/igl/ramer_douglas_peucker.cpp
+++ b/include/igl/ramer_douglas_peucker.cpp
@@ -65,7 +65,7 @@ IGL_INLINE void igl::ramer_douglas_peucker(
   };
   simplify(0,n-1);
   igl::find(I,J);
-  S = P(J.derived(),Eigen::all);
+  S = P(J.derived(),Eigen::indexing::all);
 }
 
 template <
@@ -119,7 +119,7 @@ IGL_INLINE void igl::ramer_douglas_peucker(
       T(t) = 0;
     }
   }
-  DerivedS SB = S(B,Eigen::all);
+  DerivedS SB = S(B,Eigen::indexing::all);
   Eigen::VectorXi MB = B.array()+1;
   for(int b = 0;b<MB.size();b++)
   {
@@ -128,7 +128,7 @@ IGL_INLINE void igl::ramer_douglas_peucker(
       MB(b) = S.rows()-1;
     }
   }
-  DerivedS SMB = S(MB,Eigen::all);
+  DerivedS SMB = S(MB,Eigen::indexing::all);
   Q = SB.array() + ((SMB.array()-SB.array()).colwise()*T.array());
 
   // Remove extra point at end
diff --git a/include/igl/remove_duplicate_vertices.cpp b/include/igl/remove_duplicate_vertices.cpp
index 9f6609cf..89a98db1 100644
--- a/include/igl/remove_duplicate_vertices.cpp
+++ b/include/igl/remove_duplicate_vertices.cpp
@@ -32,7 +32,7 @@ IGL_INLINE void igl::remove_duplicate_vertices(
     DerivedV rV,rSV;
     round((V/(epsilon)).eval(),rV);
     unique_rows(rV,rSV,SVI,SVJ);
-    SV = V(SVI.derived(),Eigen::all);
+    SV = V(SVI.derived(),Eigen::indexing::all);
   }else
   {
     unique_rows(V,SV,SVI,SVJ);
diff --git a/include/igl/remove_unreferenced.cpp b/include/igl/remove_unreferenced.cpp
index edf01755..dba27b50 100644
--- a/include/igl/remove_unreferenced.cpp
+++ b/include/igl/remove_unreferenced.cpp
@@ -46,7 +46,7 @@ IGL_INLINE void igl::remove_unreferenced(
   NF = F;
   std::for_each(NF.data(),NF.data()+NF.size(),
     [&I](typename DerivedNF::Scalar & a){a=I(a);});
-  NV = V(J.derived(),Eigen::all);
+  NV = V(J.derived(),Eigen::indexing::all);
 }
 
 template <
diff --git a/include/igl/resolve_duplicated_faces.cpp b/include/igl/resolve_duplicated_faces.cpp
index a3ce9fd7..b75aa180 100644
--- a/include/igl/resolve_duplicated_faces.cpp
+++ b/include/igl/resolve_duplicated_faces.cpp
@@ -81,7 +81,7 @@ IGL_INLINE void igl::resolve_duplicated_faces(
   const size_t num_kept = kept_faces.size();
   J.resize(num_kept, 1);
   std::copy(kept_faces.begin(), kept_faces.end(), J.data());
-  F2 = F1(J.derived(),Eigen::all);
+  F2 = F1(J.derived(),Eigen::indexing::all);
 }
 
 #ifdef IGL_STATIC_LIBRARY
diff --git a/include/igl/slice.cpp b/include/igl/slice.cpp
index 54093ccf..f3c856f9 100644
--- a/include/igl/slice.cpp
+++ b/include/igl/slice.cpp
@@ -121,10 +121,8 @@ IGL_INLINE void igl::slice(
     const Eigen::DenseBase<DerivedC> &C,
     Eigen::PlainObjectBase<DerivedY> &Y)
 {
-#ifndef NDEBUG
   int xm = X.rows();
   int xn = X.cols();
-#endif
   int ym = R.size();
   int yn = C.size();
 
diff --git a/include/igl/slice.h b/include/igl/slice.h
index e17701c6..9ca0e9ae 100644
--- a/include/igl/slice.h
+++ b/include/igl/slice.h
@@ -8,6 +8,7 @@
 #ifndef IGL_SLICE_H
 #define IGL_SLICE_H
 #include "igl_inline.h"
+#include <cassert>
 
 /// @file slice.h
 ///
@@ -19,11 +20,11 @@
 /// | igl                        | Eigen v3.4                         |
 /// |-------------------0--------|------------------------------------| |
 /// `igl::slice(X,I,J,Y)`      | `Y = X(I,J)`                       | |
-/// `igl::slice(X,I,1,Y)`      | `Y = X(I,Eigen::all)`              | |
-/// `igl::slice(X,J,2,Y)`      | `Y = X(Eigen::all,J)`              | |
+/// `igl::slice(X,I,1,Y)`      | `Y = X(I,Eigen::indexing::all)`              | |
+/// `igl::slice(X,J,2,Y)`      | `Y = X(Eigen::indexing::all,J)`              | |
 /// `igl::slice_into(Z,I,J,X)` | `X(I,J) = Z`                       | |
-/// `igl::slice_into(Z,I,1,X)` | `X(I,Eigen::all) = Z`              | |
-/// `igl::slice_into(Z,J,2,X)` | `X(Eigen::all,J) = Z`              | |
+/// `igl::slice_into(Z,I,1,X)` | `X(I,Eigen::indexing::all) = Z`              | |
+/// `igl::slice_into(Z,J,2,X)` | `X(Eigen::indexing::all,J) = Z`              | |
 /// `igl::slice_mask(X,M,N,Y)` | `Y = X(igl::find(M),igl::find(N))` | | _not
 /// available_            | `X(igl::find(M),igl::find(N)) = Z` |
 ///
diff --git a/include/igl/sort_triangles.cpp b/include/igl/sort_triangles.cpp
index 300c0f98..156637ad 100644
--- a/include/igl/sort_triangles.cpp
+++ b/include/igl/sort_triangles.cpp
@@ -46,7 +46,7 @@ IGL_INLINE void igl::sort_triangles(
       MV.template cast<Scalar>().transpose()*
        P.template cast<Scalar>().transpose().eval().col(2));
   sort(D,1,false,sD,I);
-  FF = F(I.derived(),Eigen::all);
+  FF = F(I.derived(),Eigen::indexing::all);
 }
 
 
diff --git a/include/igl/spectra/eigs.cpp b/include/igl/spectra/eigs.cpp
index b5caaf30..10bd0ba6 100644
--- a/include/igl/spectra/eigs.cpp
+++ b/include/igl/spectra/eigs.cpp
@@ -127,7 +127,7 @@ IGL_INLINE bool igl::spectra::eigs(
 
   Eigen::VectorXi I;
   igl::sort( Eigen::VectorXd(S), 1, false, S, I);
-  U = U(Eigen::all,I).eval();
+  U = U(Eigen::indexing::all,I).eval();
   return true;
 }
 
diff --git a/include/igl/split_nonmanifold.cpp b/include/igl/split_nonmanifold.cpp
index 761ba6e5..e3659633 100644
--- a/include/igl/split_nonmanifold.cpp
+++ b/include/igl/split_nonmanifold.cpp
@@ -64,9 +64,9 @@ IGL_INLINE void igl::split_nonmanifold(
   igl::ismember_rows(FE,FE_flip,I,J);
   // Just keep those find
   const auto II = igl::find(I);
-  Eigen::MatrixXi EI = E(II,Eigen::all);
+  Eigen::MatrixXi EI = E(II,Eigen::indexing::all);
   Eigen::VectorXi JI = J(II);
-  Eigen::MatrixXi EJI = E(JI,Eigen::all);
+  Eigen::MatrixXi EJI = E(JI,Eigen::indexing::all);
 
   Eigen::MatrixXi EJI_flip = EJI.rowwise().reverse();
   // Build adjacency matrix
@@ -155,7 +155,7 @@ IGL_INLINE void igl::split_nonmanifold(
   Eigen::PlainObjectBase <DerivedSVI> & SVI)
 {
   igl::split_nonmanifold(F,SF,SVI);
-  SV = V(SVI.derived(),Eigen::all);
+  SV = V(SVI.derived(),Eigen::indexing::all);
 }
 
 #ifdef IGL_STATIC_LIBRARY
diff --git a/include/igl/straighten_seams.cpp b/include/igl/straighten_seams.cpp
index 3e1ccc85..d9469808 100644
--- a/include/igl/straighten_seams.cpp
+++ b/include/igl/straighten_seams.cpp
@@ -74,8 +74,8 @@ IGL_INLINE void igl::straighten_seams(
   Array<bool,Dynamic,1>vBT = Map<Array<bool,Dynamic,1> >(BT.data(),BT.size(),1);
   Array<bool,Dynamic,1>vBF = Map<Array<bool,Dynamic,1> >(BF.data(),BF.size(),1);
   const auto vBT_i = igl::find(vBT);
-  MatrixX2I OF = EF(vBT_i,Eigen::all);
-  OT = EF(vBT_i,Eigen::all);
+  MatrixX2I OF = EF(vBT_i,Eigen::indexing::all);
+  OT = EF(vBT_i,Eigen::indexing::all);
   VectorXi OFMAP = EFMAP(vBT_i);
   // Two boundary edges on the texture-mapping are "equivalent" to each other on
   // the 3D-mesh if their 3D-mesh vertex indices match
@@ -115,7 +115,7 @@ IGL_INLINE void igl::straighten_seams(
   assert( (M.array() == 1).all() );
   VectorXi DT;
   // Map counts onto texture-vertices
-  DT = DV(I,Eigen::all);
+  DT = DV(I,Eigen::indexing::all);
   // Boundary in 3D && UV
   Array<bool,Dynamic,1> BTF = vBF(igl::find(vBT));
 
@@ -235,14 +235,14 @@ IGL_INLINE void igl::straighten_seams(
     {
       case 1:
         {
-          MatrixX2I OTIc = OT(Ic,Eigen::all);
+          MatrixX2I OTIc = OT(Ic,Eigen::indexing::all);
           edges_to_path(OTIc,vpath,epath,eend);
           Array<bool,Dynamic,1> SVvpath = SV(vpath);
           assert(
             (vpath(0) != vpath(vpath.size()-1) || !SVvpath.any()) && 
             "Not dealing with 1-loops touching 'sharp' corners");
           // simple open boundary
-          MatrixX2S PI = VT(vpath,Eigen::all);
+          MatrixX2S PI = VT(vpath,Eigen::indexing::all);
           const Scalar bbd = 
             (PI.colwise().maxCoeff() - PI.colwise().minCoeff()).norm();
           // Do not collapse boundaries to fewer than 3 vertices
@@ -256,7 +256,7 @@ IGL_INLINE void igl::straighten_seams(
           {
             MatrixX2S UPI,UTvpath;
             ramer_douglas_peucker(PI,eff_tol*bbd,UPI,UIc,UTvpath);
-            UT(vpath,Eigen::all) = UTvpath;
+            UT(vpath,Eigen::indexing::all) = UTvpath;
             if(!is_closed || allow_boundary_collapse)
             {
               break;
@@ -294,8 +294,8 @@ IGL_INLINE void igl::straighten_seams(
           }
           Array<bool,Dynamic,1> flipped;
           {
-            MatrixX2I OFIc = OF(Ic,Eigen::all);
-            MatrixX2I OFIcc = OF(Icc,Eigen::all);
+            MatrixX2I OFIc = OF(Ic,Eigen::indexing::all);
+            MatrixX2I OFIcc = OF(Icc,Eigen::indexing::all);
             Eigen::VectorXi XOR,IA,IB;
             setxor(OFIc,OFIcc,XOR,IA,IB);
             assert(XOR.size() == 0);
@@ -309,7 +309,7 @@ IGL_INLINE void igl::straighten_seams(
             vUE.push_back({OT(Icc(0),flipped(0)?1:0),OT(Icc(0),flipped(0)?0:1)});
           }else
           {
-            MatrixX2I OTIc = OT(Ic,Eigen::all);
+            MatrixX2I OTIc = OT(Ic,Eigen::indexing::all);
             edges_to_path(OTIc,vpath,epath,eend);
             // Flip endpoints if needed
             for(int e = 0;e<eend.size();e++)if(flipped(e))eend(e)=1-eend(e);
@@ -335,8 +335,8 @@ IGL_INLINE void igl::straighten_seams(
             Matrix<Scalar,Dynamic,Dynamic> UPI,SI;
             VectorXi UIc;
             ramer_douglas_peucker(PI,tol*bbd,UPI,UIc,SI);
-            UT(vpath,Eigen::all) = SI.leftCols (VT.cols());
-            UT(vpathc,Eigen::all) = SI.rightCols(VT.cols());
+            UT(vpath,Eigen::indexing::all) = SI.leftCols (VT.cols());
+            UT(vpathc,Eigen::indexing::all) = SI.rightCols(VT.cols());
             for(int i = 0;i<UIc.size()-1;i++)
             {
               vUE.push_back({vpath(UIc(i)),vpath(UIc(i+1))});
diff --git a/include/igl/triangle/cdt.cpp b/include/igl/triangle/cdt.cpp
index d5c7a3fc..b1e1a258 100644
--- a/include/igl/triangle/cdt.cpp
+++ b/include/igl/triangle/cdt.cpp
@@ -35,7 +35,7 @@ IGL_INLINE void igl::triangle::cdt(
   igl::remove_duplicate_vertices(DerivedWV(WV),DerivedWE(WE),1e-10,WV,_,J,WE);
   // Remove degenerate edges
   const Eigen::Array<bool,Eigen::Dynamic,1> keep = (WE.array().col(0) != WE.array().col(1));
-  WE = WE(keep,Eigen::all).eval();
+  WE = WE(keep,Eigen::indexing::all).eval();
   // c flag must be present
   igl::triangle::triangulate(DerivedWV(WV),WE,DerivedWV(),flags,WV,WF);
   Eigen::VectorXi UJ;
diff --git a/include/igl/triangle/scaf.cpp b/include/igl/triangle/scaf.cpp
index d75f2f15..6d29ade6 100644
--- a/include/igl/triangle/scaf.cpp
+++ b/include/igl/triangle/scaf.cpp
@@ -433,7 +433,7 @@ IGL_INLINE void build_surface_linear_system(const SCAFData &s, Eigen::SparseMatr
   }
   else
   {
-    MatrixXd bnd_pos = s.w_uv(bnd_ids, Eigen::all);
+    MatrixXd bnd_pos = s.w_uv(bnd_ids, Eigen::indexing::all);
 
     ArrayXi known_ids(bnd_ids.size() * dim);
     ArrayXi unknown_ids((v_n - bnd_ids.rows()) * dim);
@@ -494,7 +494,7 @@ IGL_INLINE void build_scaffold_linear_system(const SCAFData &s, Eigen::SparseMat
 
   auto bnd_n = bnd_ids.size();
   IGL_ASSERT(bnd_n > 0);
-  MatrixXd bnd_pos = s.w_uv(bnd_ids, Eigen::all);
+  MatrixXd bnd_pos = s.w_uv(bnd_ids, Eigen::indexing::all);
 
   ArrayXi known_ids(bnd_ids.size() * dim);
   ArrayXi unknown_ids((v_n - bnd_ids.rows()) * dim);
@@ -565,7 +565,7 @@ IGL_INLINE void solve_weighted_arap(SCAFData &s, Eigen::MatrixXd &uv)
   const auto v_n = s.v_num;
   const auto bnd_n = bnd_ids.size();
   assert(bnd_n > 0);
-  MatrixXd bnd_pos = s.w_uv(bnd_ids, Eigen::all);
+  MatrixXd bnd_pos = s.w_uv(bnd_ids, Eigen::indexing::all);
 
   ArrayXi known_ids(bnd_n * dim);
   ArrayXi unknown_ids((v_n - bnd_n) * dim);
diff --git a/include/igl/uniformly_sample_two_manifold.cpp b/include/igl/uniformly_sample_two_manifold.cpp
index 421ae5b1..a905b7e3 100644
--- a/include/igl/uniformly_sample_two_manifold.cpp
+++ b/include/igl/uniformly_sample_two_manifold.cpp
@@ -88,7 +88,7 @@ IGL_INLINE void igl::uniformly_sample_two_manifold(
   // First get sampling as best as possible on mesh
   uniformly_sample_two_manifold_at_vertices(W,k,push,S);
   verbose("Lap: %g\n",get_seconds()-start);
-  WS = W(S,Eigen::all);
+  WS = W(S,Eigen::indexing::all);
   //cout<<"WSmesh=["<<endl<<WS<<endl<<"];"<<endl;
 
 //#ifdef EXTREME_VERBOSE
@@ -361,7 +361,7 @@ IGL_INLINE void igl::uniformly_sample_two_manifold_at_vertices(
   // Remove corners, which better be at top
   S = S.segment(W.cols(),k).eval();
 
-  MatrixXd WS = W(S,Eigen::all);
+  MatrixXd WS = W(S,Eigen::indexing::all);
   //cout<<"WSpartition=["<<endl<<WS<<endl<<"];"<<endl;
 
   // number of vertices
